libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt

```diff
@@ -53,8 +53,10 @@ internal class RouteRefreshController(
 
     private fun supportsRefresh(route: DirectionsRoute?): Boolean {
         val isTrafficProfile = route?.routeOptions()
-            ?.profile()?.equals(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
-        return isTrafficProfile == true
+            ?.profile()?.equals(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC) ?: false
+        val isRouteOverviewFull = route?.routeOptions()
+            ?.overview()?.equals(DirectionsCriteria.OVERVIEW_FULL) ?: false
+        return isTrafficProfile && isRouteOverviewFull
```

Comment:
Note that either congestion or maxspeed annotations also need to be set for refresh to be supported.

Reply:
ah ok cool, added another check